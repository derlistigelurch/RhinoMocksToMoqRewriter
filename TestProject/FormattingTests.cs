//  Copyright (c) rubicon IT GmbH
//
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
//

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata;
using NUnit.Framework;
using Rhino.Mocks;
using TestProject.TestClasses;
using TestProject.TestInterfaces;

namespace TestProject
{
  [TestFixture]
  public class FormattingTests
  {
    private IValidationRuleCollectorProvider _involvedTypeProviderStub;
    private IValidationRuleCollectorProvider _validationRuleCollectorProviderMock1;
    private IValidationRuleCollectorProvider _validationRuleCollectorProviderMock2;
    private IValidationRuleCollectorProvider _validationRuleCollectorProviderMock3;
    private IValidationRuleCollectorProvider _validationRuleCollectorProviderMock4;
    private AggregatingValidationRuleCollectorProvider _validationRuleCollectorProvider;
    private MockRepository _mockRepository;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo1;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo2;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo3;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo4;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo5;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo6;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo7;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo8;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo9;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo10;

    [SetUp]
    public void SetUp ()
    {
      _involvedTypeProviderStub = MockRepository.GenerateStub<IValidationRuleCollectorProvider>();

      _mockRepository = new MockRepository();

      _validationRuleCollectorProviderMock1 = _mockRepository.StrictMock<IValidationRuleCollectorProvider>();
      _validationRuleCollectorProviderMock2 = _mockRepository.StrictMock<IValidationRuleCollectorProvider>();
      _validationRuleCollectorProviderMock3 = _mockRepository.StrictMock<IValidationRuleCollectorProvider>();
      _validationRuleCollectorProviderMock4 = _mockRepository.StrictMock<IValidationRuleCollectorProvider>();

      _validationRuleCollectorProvider = new AggregatingValidationRuleCollectorProvider (
          _involvedTypeProviderStub,
          new[]
          {
              _validationRuleCollectorProviderMock1, _validationRuleCollectorProviderMock2,
              _validationRuleCollectorProviderMock3, _validationRuleCollectorProviderMock4
          });

      var validationRuleCollector = MockRepository.GenerateStub<IValidationRuleCollector>();
      _validationRuleCollectorInfo1 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo2 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo3 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo4 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo5 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo6 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo7 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo8 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo9 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo10 = new ValidationRuleCollectorInfo (validationRuleCollector, typeof (AggregatingValidationRuleCollectorProvider));
    }

    [Test]
    public void AnyTest ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
      var typeGroup2 = new[] { typeof (ITestInterface) };
      var typeGroup3 = new[] { typeof (Customer) };

      using (_mockRepository.Ordered())
      {
        _validationRuleCollectorProviderMock1
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Return (new[] { new[] { _validationRuleCollectorInfo1 }, new[] { _validationRuleCollectorInfo2 } });
        _validationRuleCollectorProviderMock2
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Return (new[] { new[] { _validationRuleCollectorInfo3 }, new[] { _validationRuleCollectorInfo4 } });

        _validationRuleCollectorProviderMock3
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Return (new[] { new[] { _validationRuleCollectorInfo5 } });
        _validationRuleCollectorProviderMock4
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());

        _validationRuleCollectorProviderMock1
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup2))
            .Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());
        _validationRuleCollectorProviderMock2
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup2))
            .Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());

        _validationRuleCollectorProviderMock3
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup2)).Return (new[] { new[] { _validationRuleCollectorInfo6 } });
        _validationRuleCollectorProviderMock4
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup2)).Return (new[] { new[] { _validationRuleCollectorInfo7 } });

        _validationRuleCollectorProviderMock1
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup3))
            .Return (new[] { new[] { _validationRuleCollectorInfo8, _validationRuleCollectorInfo9 }, new[] { _validationRuleCollectorInfo10 } });
        _validationRuleCollectorProviderMock2
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup3))
            .Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());

        _validationRuleCollectorProviderMock3
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup3)).Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());
        _validationRuleCollectorProviderMock4
            .Expect (mock => mock.GetValidationRuleCollectors (typeGroup3)).Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());
      }

      var a = new[] { _validationRuleCollectorProviderMock1, _validationRuleCollectorProviderMock2, _validationRuleCollectorProviderMock3 };

      Assert.That (42, Is.GreaterThan (TimeSpan.FromMilliseconds (5.0))); // total
      Assert.That (21, Is.GreaterThan (TimeSpan.FromMilliseconds (5.0))); // since last checkpoint
    }

    [Test]
    public void Rewrite_MethodDeletion ()
    {
      _mockRepository.ReplayAll();

      _validationRuleCollectorProviderMock1.Replay();
    }

    [Test]
    public void Rewrite_MethodDeletion_123 ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
      _validationRuleCollectorProviderMock1.Stub (
              mock => mock.GetValidationRuleCollectors (typeGroup1)).Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());
    }

    [Test]
    public void Rewrite_MethodDeletion_WhenCalled_formatting ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
      _validationRuleCollectorProviderMock1
          .Expect (mock => mock.GetValidationRuleCollectors (typeGroup1))
          .Return (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
          .WhenCalled (
              mi =>
              {
                var a = mi.Arguments[0];
              });
    }

    [Test]
    public void Rewrite_MethodDeletion_abc ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };

      _validationRuleCollectorProviderMock1.Replay();
    }

    [Test]
    public void d ()
    {
      _mockRepository = new MockRepository();

      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };

      _validationRuleCollectorProviderMock1.Stub (_ => _.PropertyValueChanging (null, null, null, null, null)).IgnoreArguments().WhenCalled (
          mi =>
          {
            var propertyDefinition = ((PropertyDefinition) mi.Arguments[2]);
            var newValue = ((string) mi.Arguments[4]);
          });
    }
  }
}