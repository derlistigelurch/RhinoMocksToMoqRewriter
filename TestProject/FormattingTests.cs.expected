//  Copyright (c) rubicon IT GmbH
//
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
//

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata;
using Moq;
using Moq.Protected;
using NUnit.Framework;
using Rhino.Mocks;
using TestProject.TestClasses;
using TestProject.TestInterfaces;
using MockRepository = Rhino.Mocks.MockRepository;

namespace TestProject
{
  [TestFixture]
  public class FormattingTests
  {
    private Mock<IValidationRuleCollectorProvider> _involvedTypeProviderStub;
    private Mock<IValidationRuleCollectorProvider> _validationRuleCollectorProviderMock1;
    private Mock<IValidationRuleCollectorProvider> _validationRuleCollectorProviderMock2;
    private Mock<IValidationRuleCollectorProvider> _validationRuleCollectorProviderMock3;
    private Mock<IValidationRuleCollectorProvider> _validationRuleCollectorProviderMock4;
    private AggregatingValidationRuleCollectorProvider _validationRuleCollectorProvider;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo1;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo2;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo3;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo4;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo5;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo6;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo7;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo8;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo9;
    private ValidationRuleCollectorInfo _validationRuleCollectorInfo10;

    [SetUp]
    public void SetUp ()
    {
      _involvedTypeProviderStub = new Mock<IValidationRuleCollectorProvider>();

      _validationRuleCollectorProviderMock1 = new Mock<IValidationRuleCollectorProvider> (MockBehavior.Strict);
      _validationRuleCollectorProviderMock2 = new Mock<IValidationRuleCollectorProvider> (MockBehavior.Strict);
      _validationRuleCollectorProviderMock3 = new Mock<IValidationRuleCollectorProvider> (MockBehavior.Strict);
      _validationRuleCollectorProviderMock4 = new Mock<IValidationRuleCollectorProvider> (MockBehavior.Strict);

      _validationRuleCollectorProvider = new AggregatingValidationRuleCollectorProvider (
          _involvedTypeProviderStub.Object,
          new[]
          {
              _validationRuleCollectorProviderMock1.Object, _validationRuleCollectorProviderMock2.Object,
              _validationRuleCollectorProviderMock3.Object, _validationRuleCollectorProviderMock4.Object
          });

      var validationRuleCollector = new Mock<IValidationRuleCollector>();
      _validationRuleCollectorInfo1 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo2 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo3 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo4 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo5 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo6 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo7 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo8 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo9 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
      _validationRuleCollectorInfo10 = new ValidationRuleCollectorInfo (validationRuleCollector.Object, typeof (AggregatingValidationRuleCollectorProvider));
    }

    [Test]
    public void AnyTest ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
      var typeGroup2 = new[] { typeof (ITestInterface) };
      var typeGroup3 = new[] { typeof (Customer) };

      var sequence = new MockSequence();
      _validationRuleCollectorProviderMock1
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Returns (new[] { new[] { _validationRuleCollectorInfo1 }, new[] { _validationRuleCollectorInfo2 } })
            .Verifiable();
      _validationRuleCollectorProviderMock2
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Returns (new[] { new[] { _validationRuleCollectorInfo3 }, new[] { _validationRuleCollectorInfo4 } })
            .Verifiable();
      _validationRuleCollectorProviderMock3
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Returns (new[] { new[] { _validationRuleCollectorInfo5 } })
            .Verifiable();
      _validationRuleCollectorProviderMock4
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup1))
            .Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
            .Verifiable();
      _validationRuleCollectorProviderMock1
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup2))
            .Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
            .Verifiable();
      _validationRuleCollectorProviderMock2
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup2))
            .Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
            .Verifiable();
      _validationRuleCollectorProviderMock3
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup2)).Returns (new[] { new[] { _validationRuleCollectorInfo6 } })
            .Verifiable();
      _validationRuleCollectorProviderMock4
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup2)).Returns (new[] { new[] { _validationRuleCollectorInfo7 } })
            .Verifiable();
      _validationRuleCollectorProviderMock1
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup3))
            .Returns (new[] { new[] { _validationRuleCollectorInfo8, _validationRuleCollectorInfo9 }, new[] { _validationRuleCollectorInfo10 } })
            .Verifiable();
      _validationRuleCollectorProviderMock2
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup3))
            .Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
            .Verifiable();
      _validationRuleCollectorProviderMock3
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup3)).Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
            .Verifiable();
      _validationRuleCollectorProviderMock4
            .InSequence (sequence)
            .Setup (mock => mock.GetValidationRuleCollectors (typeGroup3)).Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
            .Verifiable();

      var a = new[] { _validationRuleCollectorProviderMock1.Object, _validationRuleCollectorProviderMock2.Object, _validationRuleCollectorProviderMock3.Object };

      Assert.That (42, Is.GreaterThan (TimeSpan.FromMilliseconds (5.0))); // total
      Assert.That (21, Is.GreaterThan (TimeSpan.FromMilliseconds (5.0))); // since last checkpoint
    }

    [Test]
    public void Rewrite_MethodDeletion ()
    {
    }

    [Test]
    public void Rewrite_MethodDeletion_123 ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
      _validationRuleCollectorProviderMock1.Setup (
              mock => mock.GetValidationRuleCollectors (typeGroup1)).Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>());
    }

    [Test]
    public void Rewrite_MethodDeletion_WhenCalled_formatting ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
      _validationRuleCollectorProviderMock1
          .Setup (mock => mock.GetValidationRuleCollectors (typeGroup1))
          .Returns (Enumerable.Empty<IEnumerable<ValidationRuleCollectorInfo>>())
          .Callback (
              (IEnumerable<Type> types) =>
              {
                var a = mi.Arguments[0];
              })
          .Verifiable();
    }

    [Test]
    public void Rewrite_MethodDeletion_abc ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };
    }

    [Test]
    public void d ()
    {
      var typeGroup1 = new[] { typeof (ITestInterface), typeof (ICollection) };

      _validationRuleCollectorProviderMock1.Setup (_ => _.PropertyValueChanging (It.IsAny<ClientTransaction>(), It.IsAny<IPropertyInformation>(), It.IsAny<IControl>(), It.IsAny<object>(), It.IsAny<object>())).Callback (
          (ClientTransaction o, IPropertyInformation o1, IControl o2, object o3, object o4) =>
          {
            var propertyDefinition = ((PropertyDefinition) mi.Arguments[2]);
            var newValue = ((string) mi.Arguments[4]);
          });
    }
  }
}